Scripts/Python/grsnWallPython.py

```diff
@@ -44,1203 +44,1181 @@
+                    ageSDL[sdl] = tuple(wall)
+                PanelSound[team].run(self.key, state='blockerOff')
+                self.SetPanelBlocker(team, id, False, onlyLight=True)
+
+    def GetNumBlockerSet(self,team):
+        ageSDL = PtGetAgeSDL()
+        if(team == kNorth):
+            sdl = "northWall"
+        elif(team == kSouth):
+            sdl = "southWall"
+
+        count = 0
+        for i in range(0,20):
+            if(ageSDL[sdl][i] != -1):
+                count += 1
+            else:
+                break
+
+        return count
+
+    def ResetWall(self,resetState=True):
+        if(self.IAmMaster()):
+            ageSDL = PtGetAgeSDL()
+            ageSDL["northWall"] = (-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,)
+            ageSDL["southWall"] = (-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,)
+
+            ageSDL["nChairOccupant"] = (-1,)
+            ageSDL["sChairOccupant"] = (-1,)
+            ageSDL["nWallPlayer"] = (-1,)
+            ageSDL["sWallPlayer"] = (-1,)
+            ageSDL["NumBlockers"] = (0,)
+
+            if(resetState):
+                ageSDL["nState"] = (kStandby,)
+                ageSDL["sState"] = (kStandby,)
+
+        for i in range(0,171):
+            NorthWall.value[i].physics.disable()
+            SouthWall.value[i].physics.disable()
+            self.SetPanelBlocker(kNorth, i, False, onlyLight=True)
+            self.SetPanelBlocker(kSouth, i, False, onlyLight=True)
+
+        self.SetPanelMode(kNorth)
+        self.SetPanelMode(kSouth)
+
+        NorthTubeExclude.clear(self.key)
+        SouthTubeExclude.clear(self.key)
+        NorthTubeClose.run(self.key,fastforward=1)
+        SouthTubeClose.run(self.key,fastforward=1)
+
+        self.oldCount[0] = 0
+        self.oldCount[1] = 0
+
+        if(eventHandler):
+            eventHandler.Reset()
+
+        self.TOCPanelLight() #TOC
+
+    def OnBackdoorMsg(self, target, param):
+        ageSDL = PtGetAgeSDL()
+        if target == "wallreset":
+            self.ResetWall()
+        elif target == "wallprep":
+            ageSDL["NumBlockers"] = (int(param),)
+        elif target == "wallstart":
+            ageSDL["nState"] = (kSetBlocker,)
+            ageSDL["sState"] = (kSetBlocker,)
+
+    def CheckBlockersValid(self,team,ageSDL):
+
+        # create shallow copy of full edge array
+        myEdges = list(edgeList)
+
+        for i in range(0,20):
+            blockerId = ageSDL[team][i]
+            if blockerId == -1:
+                # reached the end of set blocker list
+                break
+            elif blockerToEdgeDict[blockerId] is not None:
+                myEdges.remove(blockerToEdgeDict[blockerId])
+            else:
+                # this blocker has no effect on edges, it's a throwaway
+                pass
+
+        adjacencyList = dict()
+        for nodeA, nodeB in myEdges:
+            adjacencyList.setdefault(nodeA,[])
+            adjacencyList.setdefault(nodeB,[])
```

Comment:
```suggestion
        adjacencyList = defaultdict(list)
        for nodeA, nodeB in myEdges:
```

Add
```python
from collections import defaultdict
```
to the top of the file with the rest of the imports.